server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -682,13 +681,6 @@ public void batchDelete(List<Pool> pools) {
         for (Pool pool : pools) {
             currentSession().delete(pool);
         }
-        for (Pool pool : pools) {
```

Comment:
Is there any way to check if pool.entitlements has been initialized? Worst case, please update the javadoc to include this information. Best case, we could guard against the issue and raise an exception so that testing would catch it if someone used it after entitlements have been initialized.